napari/components/layerlist.py

```diff
@@ -19,7 +19,153 @@
     from npe2.manifest.io import WriterContribution
 
 
-class LayerList(SelectableEventedList[Layer]):
+class _LayerListMixin:
+    def __newlike__(self, data):
+        return type(self)(data)
+
+    def _on_selection_changed(self, event):
+        # This method is a temporary workaround to the fact that the Points
+        # layer needs to know when its selection state changes so that it can
+        # update the highlight state.  This (and the layer._on_selection
+        # method) can be removed once highlighting logic has been removed from
+        # the layer model.
+        for layer in event.added:
+            layer._on_selection(True)
+        for layer in event.removed:
+            layer._on_selection(False)
```

Comment:
We should remove those other deprecated methods - i believe here https://github.com/napari/napari/blob/6a3e11aa717e7928a0a5a3c7693577729a466ef1/napari/layers/base/base.py#L317-L330
and then drop this

Reply:
ah just realized this method already existed and is just moving

Reply:
yeah, we can't actually remove this yet, because we still use them internally here:
https://github.com/napari/napari/blob/6a3e11aa717e7928a0a5a3c7693577729a466ef1/napari/layers/base/base.py#L1679-L1689